qiskit/circuit/__init__.py

```diff
@@ -875,7 +875,9 @@ def power(self, exponent: float):
             # Also we have an efficient representation of power.
             return RXZGate(exponent * self.params[0])
 
-        def __array__(self, dtype=None):
+        def __array__(self, dtype=None, copy=None):
+            if copy is False:
+                raise ValueError("cannot produce a matrix without calculation")
```

Comment:
Should we expand the error message by one sentence here? I think it is hard to parse what is going on without the context of this PR...

Reply:
In theory, this message would only appear if a user does something like `np.array(gate, copy=False)`, so there's context from the call site that the stack trace would highlight.

Would "Unable to avoid copy while creating an array as requested" work better for you?  Looks like that's the error message that Numpy gives.

Reply:
I think that would sound better to me, thanks! :-)

Reply:
Done in c4103b7744.